src/main/java/james/storage/Storage.java

```diff
@@ -0,0 +1,93 @@
+package james.storage;
+
+import james.JamesException;
+import james.task.*;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * The storage of a taskList.
+ * The class allows reading from the file and writing to the file.
+ * Creates the file if it does not exist.
+ * The file is stored in the data folder as data/james.txt.
+ */
+public class Storage {
+
+    String filePath = "data/james.txt";
+    File file;
+
+    /**
+     * Creates the storage object.
+     * @throws JamesException If the file cannot be created.
+     */
+    public Storage() throws JamesException
+    {
+        try {
+            file = new File(filePath);
+            if (!file.exists()) {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new JamesException("File could not be created.");
+        }
+    }
+
+    /**
+     * Reads the file and loads the task list
+     * @return The task list.
+     * @throws JamesException If the file cannot be read.
+     */
+    public TaskList loadFile() throws JamesException {
+        TaskList taskList = new TaskList();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNextLine()) {
+                String[] task = sc.nextLine().split(" \\| ");
+                switch (task[0]) {
+                    case "T":
```

Comment:
Should there be an indentation for case block of codes?

Reply:
edited to have no indentation for cases